File: peek_plugin_diagram/_private/tuples/LocationIndexTuple.py
from datetime import datetime

from peek_plugin_diagram._private.PluginNames import diagramTuplePrefix
from vortex.Tuple import addTupleType, TupleField, Tuple


@addTupleType
class LocationIndexTuple(Tuple):
    __tupleType__ = diagramTuplePrefix + "LocationIndexTuple"

    modelSetKey: str = TupleField()
    indexBucket: str = TupleField()

    # The compressed (deflated) json string.
    blobData: str = TupleField()
    lastUpdate: datetime = TupleField()

    # A compressed payload of... this tuple, it's a little recursive.
    # This removes the need for the client to convert the
    # tuple to a vortexMsg in saveTuple
    encodedThisTuple : bytes = TupleField()

File: peek_plugin_diagram/plugin-module/_private/tuples/LocationIndexTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {diagramTuplePrefix} from "@peek/peek_plugin_diagram/_private";


@addTupleType
export class LocationIndexTuple extends Tuple {
    public static readonly tupleName = diagramTuplePrefix + "LocationIndexTuple";

    modelSetKey: string;
    indexBucket: string;

    // The compressed (deflated) json string.
    blobData: string | null;
    lastUpdate: string;

    // A compressed payload of... this tuple, it's a little recursive.
    // This removes the need for the client to convert the
    // tuple to a vortexMsg in saveTuple
    encodedThisTuple: string;


    constructor() {
        super(LocationIndexTuple.tupleName)
    }
}
